engine/src/mac-av-player.mm

```diff
@@ -437,8 +455,12 @@ - (void)setPlayer:(AVPlayer *)player
     }
 }
 
+#include <pthread.h>
+
 void MCAVFoundationPlayer::HandleCurrentTimeChanged(void)
 {
+    MCAssert(pthread_main_np() != 0);
```

Comment:
This assert should go in:
```

void MCPlatformCallbackSendPlayerFrameChanged(MCPlatformPlayerRef player);
void MCPlatformCallbackSendPlayerMarkerChanged(MCPlatformPlayerRef player, MCPlatformPlayerDuration time);
void MCPlatformCallbackSendPlayerCurrentTimeChanged(MCPlatformPlayerRef player);
void MCPlatformCallbackSendPlayerFinished(MCPlatformPlayerRef player);
void MCPlatformCallbackSendPlayerBufferUpdated(MCPlatformPlayerRef player);
```

Reply:
should be done

Reply:
Since the assert is now in a more appropriate place, this assert can be removed.